nusight2/src/client/components/checkbox_tree/checkbox_tree.stories.tsx

```diff
@@ -0,0 +1,153 @@
+      return new ToggleTree(opts)
+    },
+  )
+
+  @computed
+  get leaf(): boolean {
+    return this.model instanceof TreeData
+  }
+
+  @computed
+  get checked(): CheckedState {
+    if (this.model instanceof TreeData) {
+      return this.model.checked
+    }
+
+    if (this.children.every(node => node.checked === CheckedState.Checked)) {
+      return CheckedState.Checked
+    }
+
+    if (this.children.every(node => node.checked === CheckedState.Unchecked)) {
+      return CheckedState.Unchecked
+    }
+
+    return CheckedState.Indeterminate
+  }
+
+  set checked(checked: CheckedState) {
+    if (this.model instanceof TreeData) {
+      this.model.checked = checked
+    } else {
+      this.children.forEach(child => {
+        child.checked = checked
+      })
+    }
+  }
+
+  @computed
+  get children(): TreeNodeModel[] {
+    if (this.model instanceof TreeData) {
+      return []
+    }
+
+    return Array.from(
+      Object.entries(this.model).map(([key, value]) =>
+        ToggleTree.of({
+          label: key,
+          model: value,
+        }),
+      ),
+    )
+  }
+}
+
+storiesOf('components/CheckboxTree', module)
+  .addDecorator(story => <div style={{ maxWidth: '320px' }}>{story()}</div>)
+  .add('renders', () => {
+    const rawData: ToggleTreeChildren = {
+      'Robot 1': {
+        'Debug Waves': {
+          'sin()': new TreeData(),
+          'cos()': new TreeData(),
+          'tan()': new TreeData(),
+        },
+        Position: {
+          x: new TreeData(),
+          y: new TreeData(),
+          z: new TreeData(),
+        },
+      },
+      'Robot 2': {
+        'Debug Waves': {
+          'sin()': new TreeData(),
+          'cos()': new TreeData(),
+          'tan()': new TreeData(),
+        },
+        Position: {
+          x: new TreeData(),
+          y: new TreeData(),
+          z: new TreeData(),
+        },
+      },
+    }
+
+    const model = observable<TreeModel>({
+      nodes: Object.entries(rawData).map(([key, value]) => {
+        return ToggleTree.of({ label: key, model: value })
+      }),
+      usePessimisticToggle: true,
+    })
+
+    // Used to check/uncheck a node in the tree
+    const onNodeCheck = action((node: TreeNodeModel) => {
+      node.checked =
+        node.checked === CheckedState.Checked ? CheckedState.Unchecked : CheckedState.Checked
+    })
+
+    // Used to expand/collapse a node in the tree
+    const onNodeExpand = action((node: TreeNodeModel) => {
+      node.expanded = !node.expanded
+    })
+
+    // Used to render the label, can be any arbitrary content, e.g. a color picker
+    // similar to the one used in the Chart view's checkbox tree.
+    const renderLabel = (node: TreeNodeModel) => {
+      return <span>{node.label}</span>
+    }
+
+    const Story = observer(() => (
+      <CheckboxTree
+        model={model}
+        onCheck={onNodeCheck}
+        onExpand={onNodeExpand}
+        renderLabel={renderLabel}
+      />
+    ))
+
+    return <Story />
```

Comment:
No observables are accessed in this observer component, so we can simplify

```suggestion
  return <CheckboxTree
      model={model}
      onCheck={onNodeCheck}
      onExpand={onNodeExpand}
      renderLabel={renderLabel}
    />
```